frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions, searchSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from './search';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean,
+}
+interface Props {
+  submissions: any,
+  dispatch: any,
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // eslint-disable-next-line react/no-unused-state
+      loading: props.submissions.submissions === undefined,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submission === undefined) {
```

Comment:
Typo; it should be `props.submissions.submissions`. Right now if you navigate back-forward it will re-fetch another page of submissions although it shouldn't.

Also, this should have been caught by TypeScript. I see you didn't correctly declare prop types, both are any. It should be the correct types.